### ISSUE TITLE
can't release the memory of Starcoder Native INT4 model in bigdl-llm

### ISSUE THREAD

**Issue by KiwiHana** [intel-analytics/BigDL#8701]
********** Switch model from  StarCoder to llama2-13b
******* loading llama2-13b
bigdl-llm: loading model from D:\PC_LLM_UI\benchmark\checkpoint\\bigdl_llm_llama2_13b_q4_0.bin
loading bigdl-llm model: format     = ggjt v3 (latest)
loading bigdl-llm model: n_vocab    = 32000
loading bigdl-llm model: n_ctx      = 512
loading bigdl-llm model: n_embd     = 5120
loading bigdl-llm model: n_mult     = 256
loading bigdl-llm model: n_head     = 40
loading bigdl-llm model: n_layer    = 40
loading bigdl-llm model: n_rot      = 128
loading bigdl-llm model: ftype      = 2 (mostly Q4_0)
loading bigdl-llm model: n_ff       = 13824
loading bigdl-llm model: n_parts    = 1
loading bigdl-llm model: model size = 13B
loading bigdl-llm model: ggml ctx size = 6595.81 MB
loading bigdl-llm model: mem required  = 8643.81 MB (+ 1608.00 MB per state)
....................................................................................................
loading bigdl-llm model: kv self size  =  400.00 MB
Traceback (most recent call last):
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\gradio\routes.py", line 442, in run_predict
    output = await app.get_blocks().process_api(
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\gradio\blocks.py", line 1392, in process_api
    result = await self.call_function(
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\gradio\blocks.py", line 1111, in call_function
    prediction = await utils.async_iteration(iterator)
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\gradio\utils.py", line 346, in async_iteration
    return await iterator.__anext__()
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\gradio\utils.py", line 339, in __anext__
    return await anyio.to_thread.run_sync(
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\gradio\utils.py", line 322, in run_sync_iterator_async
    return next(iterator)
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\gradio\utils.py", line 691, in gen_wrapper
    yield from f(*args, **kwargs)
  File "D:\PC_LLM_UI\LLM_demo_v0.61.py", line 140, in predict
    model = load(model_path=model_all_local_path + "\\bigdl_llm_llama2_13b_q4_0.bin",
  File "D:\PC_LLM_UI\LLM_demo_v0.61.py", line 114, in load
    llm = BigdlNativeForCausalLM.from_pretrained(
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\bigdl\llm\transformers\modelling_bigdl.py", line 65, in from_pretrained
    return Llama(model_path=ggml_model_path, **kwargs)
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\bigdl\llm\ggml\model\llama\llama.py", line 207, in __init__
    self.ctx = llama_cpp.llama_init_from_file(
  File "C:\Users\BigDL-LLM\miniconda3\envs\ui_llm\lib\site-packages\bigdl\llm\ggml\model\llama\llama_cpp.py", line 294, in llama_init_from_file
    return _lib.llama_init_from_file(path_model, pointer(params))
OSError: [WinError -529697949] Windows Error 0xe06d7363

bigdl-llm version: 0807
code：
```
def load(model_path, model_family, n_threads):
    from bigdl.llm.transformers import BigdlNativeForCausalLM
    llm = BigdlNativeForCausalLM.from_pretrained(
        pretrained_model_name_or_path=model_path,
        model_family=model_family,
        n_threads=n_threads)
    return llm


def predict(input, function, chatbot, max_length, top_p, temperature, history, model_select):
    global model_name,model_all_local_path
    input = parse_text(input)
    if model_select != model_name:
        print("********** Switch model from ",model_name,"to",model_select)
        model_name = model_select
        global model,tokenizer
        del model
        gc.collect()

        model_name_local = model_all_local_path + model_name
        stm = time.time()
        if model_name == "chatglm2-6b":
            print("******* loading chatglm2-6b")
            model = AutoModel.load_low_bit(model_name_local,trust_remote_code=True)
            tokenizer = AutoTokenizer.from_pretrained(model_name_local,trust_remote_code=True)
            model = model.eval()
        elif model_name == "llama2-13b":
            print("******* loading llama2-13b")
            model = load(model_path=model_all_local_path + "\\bigdl_llm_llama2_13b_q4_0.bin",
                    model_family='llama',
                    n_threads=20)
        elif model_name == "StarCoder":
            print("******* loading StarCoder")
            model = load(model_path=model_all_local_path + "\\bigdl_llm_starcoder_q4_0.bin",
                    model_family='starcoder',
                    n_threads=20)  
        print("********** model load time (s)= ", time.time() - stm)     
```

**Comment by cyita** [comment]
This error always occurs when the .bin is damaged or a mismatch between the code used for quantization and the inference.
Please re-quantize the model and try again.
If the problem persists, please kindly share your CPU model with us to reproduce.

**Comment by KiwiHana** [comment]
Hi Yina, 

The problem is StarCoder model keep memory in bigdl of model(prompt, stream=True), we can't release memory of StarCoder model.

the log of code:

model_name = "llama2-13b"
******************* after swtich model memory used: 6898.03125 MB
******************* del model memory used: 281.6328125 MB

model_name = "StarCoder"
******************* after swtich model memory used: 10016.6875 MB
******************* del model memory used: 10234.96875 MB

```
# LLama2 starcoder load 
def load(model_path, model_family, n_threads):
    from bigdl.llm.transformers import BigdlNativeForCausalLM
    llm = BigdlNativeForCausalLM.from_pretrained(
        pretrained_model_name_or_path=model_path,
        model_family=model_family,
        n_threads=n_threads)
    return llm

# 显示当前 python 程序占用的内存大小
def show_memory_info(hint):
    pid = os.getpid()
    p = psutil.Process(pid)

    info = p.memory_full_info()
    memory = info.uss / 1024. / 1024
    print('******************* {} memory used: {} MB'.format(hint, memory))

model_name = "llama2-13b"

if model_name == "llama2-13b":
    model = load(model_path=model_all_local_path + "\\bigdl_llm_llama2_13b_q4_0.bin",
            model_family='llama',
            n_threads=20)
elif model_name == "StarCoder":
    model = load(model_path=model_all_local_path + "\\bigdl_llm_starcoder_q4_0.bin",
            model_family='starcoder',
            n_threads=20)  

response = ""

print("model_name = ", model_name)
show_memory_info('after swtich model')  

for chunk in model(prompt, stream=True):
    response += chunk['choices'][0]['text']

del model
gc.collect()
show_memory_info('del model')

```

**Comment by cyita** [comment]
OS & CPU info: `13900H windows`

**Comment by jason-dai** [comment]
@MeouSker77 please take a look

**Comment by MeouSker77** [comment]
fixed in #8728 and https://github.com/intel-analytics/llm.cpp/pull/49, waiting for verification

**Comment by shane-huang** [comment]
fix verified.
